### DISCUSSION TITLE
Address in SIRCS sending always 0x0, despite different definition

### DISCUSSION THREAD

**Discussion by BerlinJoker** [ukw100/IRMP#91] [Q&A] [answered]
Hi, let me start with saying that I really really appreciate the amount of effort that must have gone (and continues to do) into IRMP. I'm not a developer, but working in the field I do have a bit of a grasp for the efforts involved.

This is my first project using IRMP, so this might just be something I'm overlooking, but I cannot nail it down.

I built an IR sender using the SimpleSender example, adjusting it where needed to send SIRCS commands. 
The problem: no matter what I define as the address for sending, what is going out in the end is always sent using 0x0 as the address. Therefore also the TV I'm trying to control doesn't react to the commands.

I first started out with building a receiver to document the IR codes of my TV remote, which worked liked a charm. Then I built a second device for sending, adjusted the SimpleSender example, but cannot get it to work. The serial output of the sender says it used the configured address (0x97 or 0x1, I tried both), but the serial output of the receiver (AllProtocols without modification) shows 0x0 as address for the incoming commands. The commands themselves and the protocol are detected correctly.

Example output:
16:55:32.356 -> P=SIRCS | 0x1 A=0x0 C=0x19
16:55:32.496 -> P=SIRCS | 0x1 A=0x0 C=0x19 R
16:55:32.542 -> P=SIRCS | 0x1 A=0x0 C=0x19 R
16:55:32.589 -> P=SIRCS | 0x1 A=0x0 C=0x19 R
16:55:32.636 -> P=SIRCS | 0x1 A=0x0 C=0x19 R
16:55:38.492 -> P=SIRCS | 0x1 A=0x1 C=0x14
16:55:38.632 -> P=SIRCS | 0x1 A=0x1 C=0x14 R
16:55:40.159 -> P=SIRCS | 0x1 A=0x1 C=0x14
16:55:40.298 -> P=SIRCS | 0x1 A=0x1 C=0x14 R

The first 5 lines come from the modified SimpleSender, the lower lines come from pressing a button on the remote of my TV. There the correct address is detected.

Here is the section from SimpleSender that I modified from the NEC code to SIRCS:

#else
    /*
     * Send sony mute
     */
    irsnd_data.protocol = IRMP_SIRCS_PROTOCOL;
    irsnd_data.address = 0x97;
    irsnd_data.command = 0x19; // The required inverse of the 8 bit command is added by the send routine.
    irsnd_data.flags = 4; // repeat frame 4 times
#endif

I tried both 
#define IRSND_IR_FREQUENCY          40000
and
#define IRSND_IR_FREQUENCY          38000
with no effect. 

Hardware:
ESP32 devkit1 boards
IR sender: https://www.az-delivery.de/products/ir-sende-modul?_pos=2&_psq=infrarot&_ss=e&_v=1.0
IR receiver: https://www.az-delivery.de/products/ir-empfanger-modul

Software:
Arduino IDE 1.8.19
IRMP 3.6.3

Pin definitions:
#define IRSND_OUTPUT_PIN     4  // D4
#define IRMP_INPUT_PIN      15  // D15

The IR diode itself sends a signal, I can verify it by checking with my phone camera, and as mentioned above: the AllProtocols receiver recognises the commands and the protocol, just not the right address.

Is there something obvious I'm missing? Grateful for any hints :)

cheers!

**Comment by ArminJo** [comment, marked as answer]
Sircs is a bit complicated, since it has [3 variants](https://github.com/Arduino-IRremote/Arduino-IRremote/blob/c6c7f17f587eabab0833fe0d048dba85251a9735/src/ir_Sony.hpp#L100).
So it is best to use IRremote library, because it is more transparent regarding the SIRCS / Sony variants.

**Reply by BerlinJoker** [reply]
That did indeed do the trick, thanks a lot! :)

IrSender.sendSony(0x1, sCommand, sRepeats, 12);
works like a charm!
